Implementation of Lottie Theme Properties (custom colors) (#4632)
unoplatform/uno: doc/articles/features/Lottie.md

* You can specify `(assembly)` in path: will be replaced by assembly name.
* AssemblyName is case insensitive, **but the resource name is**.

## Theme Properties

As Microsoft documented in there [Release Notes for Lottie-Windows v6.1.0](https://github.com/windows-toolkit/Lottie-Windows/releases/tag/v6.1.0), there's a new feature called _Theme property binding_. This allows to dynamically change a value (usually a color) at runtime.

To use this feature on Windows, you need to generate code with the [`LottieGen` tool](https://docs.microsoft.com/en-us/windows/communitytoolkit/animations/lottie-scenarios/getting_started_codegen). On Uno supported platforms, it's available at runtime using the `ThemableLottieVisualSource` instead of the classic `LottieAnimatedVisualSource`.

Here's **how to use this feature**:

> 1. Add a _css-like_ to your Lottie shape like this - put this in the name of the shape. That means the `nm` property in the json-generated file.

Author: ```suggestion
1. Add a _css-like_ declaration to your Lottie shape like this - put this in the name of the shape. That means the `nm` property in the json-generated file.
```

> 1. Add a _css-like_ to your Lottie shape like this - put this in the name of the shape. That means the `nm` property in the json-generated file.

Author: ```suggestion
1. Add a _css-like_ declaration to your Lottie shape like this then put this in the name of the shape. That means the `nm` property in the json-generated file.
```

   ``` css
   { Color: var(MyColor); }
   ```

   In that case, it will create a property (variable) named `MyColor`. It is possible to reuse the same property name on many layers in the same lottie file.

2. Use it that way:

   ``` xml
   <winui:AnimatedVisualPlayer
       x:Name="player"
       AutoPlay="true">
   
       <lottie:ThemableLottieVisualSource
           x:Name="animation"
           UriSource="ms-appx:///Assets/Lottie/CheckBoxAnimation.json" />
   </winui:AnimatedVisualPlayer>
   ```

3. Change the property in your code:

   ```csharp
   animation.SetColorThemeProperty("MyColor", Color.FromArgb(1, 255, 0, 0));
   ```

**Notes**:

> * Changing a property will force the player to reload its animation. Not only the animation will be restarted, but it can also big heavy on the CPU. Animation those values should be avoided.

Reviewer: ```suggestion
* Changing a property will force the player to reload its animation. Not only the animation will be restarted, but it can also be heavy on the CPU. Animating these properties should be avoided.
```
